src/partial-templates/partial-templates.service.ts

```diff
@@ -18,92 +18,223 @@ export class PartialTemplatesService {
    * @returns {Promise<PartialTemplate>} - The updated partialTemplate
    */
-  async update(
-    id: number,
-    updatePartialTemplateDto: UpdatePartialTemplateDto,
-  ): Promise<PartialTemplate> {
-    // Ensures the partial template exists before updating
-    await this.findOne(id);
-    return this.prisma.partialTemplate.update({
-      data: { ...updatePartialTemplateDto } as any,
-      where: {
-        id,
-      },
-    });
+  async update(id: number, updatePartialTemplateDto: UpdatePartialTemplateDto): Promise< any> {
+    try { 
+      await this.validateId(id);
+      await this.validateForeignKeys(updatePartialTemplateDto);
+      await this.validateTotalByPosition(updatePartialTemplateDto);
+
+      const updatePartialTemplate = await this.prisma.partialTemplate.update({
+        data: { ...updatePartialTemplateDto} as any,
+        where: {
+          id
+        }
+      });
+      return {
+        message: 'Plantilla parcial actualizada',
+        errror: null,
+        data: updatePartialTemplate,
+      };
+    } catch (error){
+      return {
+        message: 'Error al actualizar la plantilla parcial',
+        error: error.message,
+        data: null,
+      };
+    }
   }
 
   /**
    * Deletes a partialTemplate by its ID
    * @param {number} id - ID of the partialTemplate to delete
    * @returns {Promise<PartialTemplate>} - The deleted partialTemplate
    */
-  async remove(id: number): Promise<{ message: string }> {
-    // Ensures the partial template exists before deleting
+  async remove(id: number): Promise<any> {
+    try{ 
     await this.findOne(id);
     await this.prisma.partialTemplate.delete({
       where: {
         id,
       },
     });
-    return { message: 'Deleted successfully' };
+    return {
+      message: 'Plantilla parcial eliminada con exito',
+      error: null,
+      data: null,
+    };
+  } catch (error){
+    return {
+      message: 'Error al eliminar la plantilla parcial',
+      error: error.message,
+      data: null,
+    };
+  }
+  }
+
+   /**
+   * Validates if partailTemplate ID exists
+   * @param {number} id - partailTemplate ID to validate
+   * @returns {Promise<Template>} - The partailTemplate with the given ID
+   * @throws {NotFoundException} - If the partailTemplate ID is not found
+   */
+  async validateId(id: number): Promise<PartialTemplate> {
+    const partialTemplate = await this.prisma.partialTemplate.findUnique({
+      where: {
+        id,
+      },
+    });
+
+    if (!partialTemplate) {
+      throw new NotFoundException('Plantilla Parcial no encontrada');
+    }
+    return partialTemplate;
   }
+
+  /**
+   * Validates if foreign keys (templateId)
+   * @param {CreatePartialTemplateDto | UpdatePartialTemplateDto} dto - Data to validate
+   * @throws {NotFoundException} - If any foreign keys is not found
+   */
+  private async validateForeignKeys( dto: CreatePartialTemplateDto | UpdatePartialTemplateDto): Promise<void> {
```

Comment:
#### Update the validateForeignKeys method because the query form is handled when you want to check multiple ID's and you are only validating one ID. In addition to the fact that the name is generic and not specific.

#### Then the update would be as follows:

```ts
  private async validateIfExistsTemplateId(id: number) {
    const template = await this.prisma.template.findUnique({
      where: {
        id,
      },
    });
    if (!template) {
      throw new NotFoundException(`Plantilla con ID ${id} no encontrada`);
    }
  }
}
```

#### You can adapt the code to the form you consider necessary.